tests/ClientTest.php

```diff
@@ -25,13 +25,41 @@
 use Symfony\Component\Panther\Client;
 use Symfony\Component\Panther\Cookie\CookieJar;
 use Symfony\Component\Panther\DomCrawler\Crawler;
+use Symfony\Component\Panther\PantherTestCase;
 use Symfony\Component\Panther\ProcessManager\ChromeManager;
 
 /**
  * @author Kévin Dunglas <dunglas@gmail.com>
  */
 class ClientTest extends TestCase
 {
+    public function testCreatePantherClientInvalidArgument(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+
+        self::createPantherClient([
+            'browser_arguments' => 'bad browser arguments data type',
+        ]);
+    }
+
+    public function testCreatePantherClientWithBrowserArgument(): void
+    {
+        $client = self::createPantherClient([
+            'browser' => PantherTestCase::CHROME,
+            'browser_arguments' => ['--window-size=1400,900'],
+        ]);
+        $this->assertInstanceOf(AbstractBrowser::class, $client);
+        $this->assertInstanceOf(WebDriver::class, $client);
+        $this->assertInstanceOf(JavaScriptExecutor::class, $client);
+        $this->assertInstanceOf(KernelInterface::class, self::$kernel);
+
+        // I'm not sure exactly how to start with a clear class, since the
+        // test file assumes the client will be present for the next tests
+        // So I'm cleaning up this way here so that we can have a default
+        // client next
+        self::$pantherClient = null;
```

Comment:
This is tricky. You're re-setting this so that `createPantherClient()` won't reuse your instance in further tests. However, it has a side effect that the `stopWebServer()` in the trait (called by `tearDownAfterClass()`) will not fire correctly.

Try `$this->stopWebServer()` instead:

```php
// stop the web server and clean up self::$pantherClient so it is not reused by other tests
$this->stopWebServer();
```

Reply:
Trying with this change now!

Reply:
wow, tests are passing now! That is great! :)